presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/DummyBookKeeper.java

```diff
@@ -44,7 +45,14 @@ public void setAllCached(String remotePath, long fileLength, long lastModified,
     }
 
     @Override
-    public boolean readData(String remotePath, long offset, int length, long fileSize, long lastModified, int clusterType)
+    public Map<String, Long> getReadRequestChainStats()
+            throws TException
+    {
+        throw new UnsupportedOperationException();
```

Comment:
This can lead to imx query failures as DummyBookKeeper is used when `hive.cache.start-server-on-coordinator=false` which is the default value. Better return an empty map here.

Reply:
Interestingly, product tests seem to pass. I think `com.qubole.rubix.core.ReadRequestChainStats#getBookKeeperStats` is resilient for that case